### PR TITLE
Improve StaticFiles performance, and make ChunkedReadFile public

### PR DESCRIPTION
**Pull request by transkatgirl**
This fixes issue #560, and also addresses some concerns brought up about ChunkedReadFile.

Sorry about the weird commit names, I was using my phone to do this.

### CODE REVIEW

File: src/server/service.rs [hunk 1]
```diff
@@ -88,7 +88,8 @@ where
         Ok(Async::Ready(()))
     }
 
-    fn call(&mut self, req: Self::Request) -> Self::Future {
+    fn call(&mut self, mut req: Self::Request) -> Self::Future {
+        let _ = req.set_nodelay(true);
```
Comment (DoumanAsh) [inline src/server/service.rs:92]: @fafhrd91 it is been removed in c674ea912691d86379d610a4794a50cef4b2feac

Do you remember why? Or was it mistake?

Comment (fafhrd91) [reply src/server/service.rs:92]: There is separate service for that

File: src/fs.rs [hunk 2]
```diff
@@ -475,12 +475,12 @@ impl<C: StaticFileConfig> Responder for NamedFile<C> {
 /// A helper created from a `std::fs::File` which reads the file
 /// chunk-by-chunk on a `CpuPool`.
 pub struct ChunkedReadFile {
-    size: u64,
-    offset: u64,
-    cpu_pool: CpuPool,
-    file: Option<File>,
-    fut: Option<CpuFuture<(File, Bytes), io::Error>>,
-    counter: u64,
+    pub size: u64,
```
Comment (fafhrd91) [inline src/fs.rs:478]: What is the reason for this change?

Comment (transkatgirl) [reply src/fs.rs:478]: ChunkedReadFile is public and shows up in the documentation, but it isn't useable in handlers, because its fields are private.

File: src/server/service.rs (hunk 1 above)
Comment (DoumanAsh) [reply src/server/service.rs:92]: You mean `StreamConfigurationService`?

File: src/fs.rs (hunk 2 above)
Comment (transkatgirl) [reply src/fs.rs:478]: So, are you okay with this change being included in the pull request?

It makes sense for ChunkedReadFile's fields to be public, as it's not useable when the fields are private. Why have ChunkedReadFile show up in the docs if you don't want people to use it?

File: src/server/service.rs (hunk 1 above)
Comment (transkatgirl) [reply src/server/service.rs:92]: Well, this is the simplest solution I found to the issue. If you want to solve the issue in a different way, I could modify the pull request.

Comment (kittyhacker101) [reply src/server/service.rs]: Well, this is the simplest solution I found to the issue. If you want to solve the issue in a different way, I could modify the pull request.

**Comment by fafhrd91** [comment]
i dont think it make sense to expose `ChunkedReadFile` atm, it is too specific to actix-web. i hid it for now. also added nodelay in master
